zookeeper-api/src/main/java/org/apache/helix/zookeeper/api/client/RealmAwareZkClient.java

```diff
@@ -175,10 +175,34 @@ void unsubscribeStateChanges(
 
   void createPersistent(String path, Object data, List<ACL> acl);
 
+  void createPersistentWithTTL(String path, long ttl);
```

Comment:
this is not specific to your review but a generic design question. 
ZK in open source will keep adding new features. As a result, do we need to keep adding similar APIs and support?

Reply:
I think this is a case-by-case situation. I'd say ideally, yes, but for this particular set of API, we have known use-cases that need these features so that's why I started adding them. In fact, they have been available in open-source Zookeeper for quite a while now, but we hadn't added them until now.